bokeh/models/sources.py

```diff
@@ -38,6 +39,10 @@ class DataSource(PlotObject):
                 hit/selected
     """)
 
+    callback = Instance(Callback, help="""
+    A callback to run in the browser whenever the selection is changed.
+    """)
+
```

Comment:
I think it's probably what we need now. 

I may be creating problems that we don't have and I'm not sure how to express it but it may be the case that we need a callback when selection has changed and a different one for when data has changed (thinking about the discussions in #2248).